includes/class-sensei-settings.php

```diff
@@ -1096,6 +1097,86 @@ public function experimental_features_saved( $old_value, $value ) {
 		}
 	}
 
+	/**
+	 * Renders the High-Performance Progress Storage feature setting.
+	 *
+	 * @param array $args The field arguments.
+	 */
+	public function render_progress_storage_feature( $args ) {
+
+		// Checkbox field.
+		$settings = $this->get_settings();
+		$key      = $args['key'];
+		$value    = $settings[ $key ];
+
+		// Disable the checkbox if we can't set time limit. That's our current limitation for running migrations.
+		$disabled                    = true;
+		$disabled_messages           = array();
+		$xdebug_enabled              = false;
+		$set_time_limit_disabled     = true;
+		$original_max_execution_time = (int) ini_get( 'max_execution_time' );
+		if ( 0 !== $original_max_execution_time && function_exists( 'set_time_limit' ) ) {
+			$set_time_limit_disabled = false;
+			// Set max execution time to 0 to check if we can set it in migrtions.
+			$disabled                   = ! set_time_limit( 0 );
+			$xdebug_enabled             = extension_loaded( 'xdebug' );
+			$current_max_execution_time = (int) ini_get( 'max_execution_time' );
+			if ( $disabled && 0 === $current_max_execution_time ) {
+				$disabled = false;
+			}
+			// Restore original max execution time.
+			set_time_limit( $original_max_execution_time );
+		}
+
+		if ( $disabled ) {
+			if ( $set_time_limit_disabled ) {
+				$disabled_messages[] = sprintf(
+					// translators: Placeholder is a link to `set_time_limit` function documentation.
+					__( '%s function is disabled in your PHP configuration.', 'sensei-lms' ),
+					'<a href="https://www.php.net/manual/en/function.set-time-limit.php" target="_blank" rel="noopener noreferrer">set_time_limit</a>'
+				);
+			}
+			if ( $xdebug_enabled ) {
+				$disabled_messages[] = __( 'Xdebug is enabled. It may affect the result of running `set_time_limit`. Please, condider disabling it.', 'sensei-lms' );
+			}
+			$disabled_messages[] = __( 'We execute long-running migrations to make the feature work.', 'sensei-lms' );
+			$disabled_messages[] = sprintf(
+				// translators: Placeholder is a link to `max_execution_time` configuration option documentation.
+				__( 'Because of that, we need to set %s for PHP scripts.', 'sensei-lms' ),
+				'<a href="https://www.php.net/manual/en/info.configuration.php#ini.max-execution-time" target="_blank" rel="noopener noreferrer">max_execution_time</a>'
+			);
+			$disabled_messages[] = sprintf(
+				// translators: Placeholder is a link to `set_time_limit` function documentation.
+				__( "Unfortunately, we can't use %s on your server and we had to disable the feature entirely.", 'sensei-lms' ),
+				'<a href="https://www.php.net/manual/en/function.set-time-limit.php" target="_blank" rel="noopener noreferrer">set_time_limit</a>'
+			);
```

Comment:
I'm thinking we can simplify the copy a bit. Maybe show only one message if xdebug is causing problems and another if the time limit is not working. For example, if xdebug is the issue, show only "Xdebug is interfering with this feature. Please, consider disabling it.". And if `set_time_limit` is not working, show "The `set_time_limit` function is disabled on your server. If you want to use this feature, please contact your hosting provider to enable it.". WDYT?

Reply:
Sorry, the copy still looks a bit odd to me (I'm not good with these things). What do you think of the following:
 
`The feature can't be enabled because the set_time_limit function is disabled in your PHP configuration.`

Reply:
Updated.
![CleanShot 2023-11-29 at 07 51 49@2x](https://github.com/Automattic/sensei/assets/329356/f76c414f-4922-422d-bbdb-0e6f43f55e97)

Reply:
Thanks and sorry for the back and forward.